### ISSUE TITLE
Can't process "]]"

### ISSUE THREAD

**Issue by bbbbx** [brendan-duncan/wgsl_reflect#53]
Code:
```wgsl
struct Uniforms {
    positions: array<vec3f, 2>,
    ids: array<i32, 2>
}

@binding(0) @group(0) var<uniform> uniforms: Uniforms ;

@compute @workgroup_size(1i, 1i, 1i)
fn main() {
    uniforms.positions[uniforms.ids[0]];
}
```

Result:
![image](https://github.com/user-attachments/assets/101468db-1228-40f4-8a15-dc6d86753713)

**Comment by brendan-duncan** [comment]
Thanks for the report, I'll get that fixed up.

**Comment by brendan-duncan** [comment]
Fixed in 1.0.8, just published.

**Comment by bbbbx** [comment]
Thank you, I can confirm the issue has been fixed
